### PR TITLE
Suppress deprecation warnings when called from bourbon

### PR DESCRIPTION
**Pull request by CupOfTea696**
### What does this PR do?

Suppresses deprecation warnings when deprecated functions or mixins from other bourbon functions or mixins. This will stop a lot of confusion when people suddenly see loads of deprecation warnings for functions or mixins they didn't use directly.

### If this is related to an existing issue, include a link to it as well.
Solves: #1001

### CODE REVIEW

File: app/assets/stylesheets/addons/_border-color.scss [hunk 1]
```diff
@@ -22,5 +22,8 @@
 /// @output `border-color`
 
 @mixin border-color($vals...) {
+  $deprecation-warnings: $output-bourbon-deprecation-warnings;
```
Comment (tysongach) [inline app/assets/stylesheets/addons/_border-color.scss]: For clarity, what about renaming this from `$deprecation-warnings` to `$user-setting-for-deprecation-warnings`?

Comment (CupOfTea696) [inline app/assets/stylesheets/addons/_border-color.scss]: Maybe just `$user-deprecation-warnings-setting`? Bit shorter.

**Comment by tysongach** [comment]
Thanks, @CupOfTea696!

Merged as https://github.com/thoughtbot/bourbon/commit/b69c1813bd4cf00b575c0350ef841696294a756e.

**Comment by tysongach** [comment]
We just released v4.3.1 which includes these changes.

**Comment by CupOfTea696** [comment]
@tysongach You're welcome, and awesome that it's released!

**Comment by JoshSmith** [comment]
@tysongach to not spam #702 I'm confirming that this is related to issues I'm seeing.

**Pull request closed without merging**